pkg/kubelet/kubelet.go

```diff
@@ -1947,6 +1947,14 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 	}
 
 	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && isPodResizeInProgress(pod, &apiPodStatus) {
+		// Before syncPod end, get the latest podStatus from CRI and refresh the pod
```

Comment:
Why is this necessary?  This is normally the responsibility of the pod handler, and refreshing status generally needs a reason.

Reply:
After calling CRI to update the Resource, there is no chance to update the Pod's ResizeStatus to completed. It need to wait until the next SyncPod, which is usually 1 minute later.

Reply:
That sounds incorrect.  If we make a CRI call, that updates the state of the object, then we should get a CRI event afterwards indicating the container runtime status changed (something should be different).  That should then trigger a subsequent SyncPod, which gives you a chance to react.  Thats <1-2s (and will be faster when the continuous pod status loop is added).

If you are seeing 1m waits (meaning you had to wait for house cleaning and sync), that means the change to the CRI either didn't generate an event, or *should* have generated an event and we didn't see it.  Can you verify for me whether the CRI update to the container actually changes the observable state of the container in such a way that it shows up in in the pleg?  Without this code, when you run SyncPod, I expect to see PLEG detect a container change.

If that's the problem, and we fix that, and it only takes 1-2s, this is no different from any other kubelet code, and we wouldn't need this change?

Reply:
@smarterclayton 

1. Agree that `updateStatus` again in the same `kubelet.SyncPod` loop in not the  right way to go. 
2. PLEG currently can not differentiate the pod just finished resizing because container state is always "running" and the hash doesn't change (hash indicate the). There's no way to generate a `ContainerChanged` pleg event. There's definitely bug in the original VPA implementation. 

I am trying to fix PLEG logic for in-place VPA now.